jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/complex/DropdownOptionsTests.java

```diff
@@ -0,0 +1,102 @@
+
+    private static final String ITEM_ONE = "Action";
+    private static final String ITEM_TWO = "Another action";
+    private static final String ITEM_THREE = "Something else here";
+    private static final String ITEM_FOUR = "Separated link";
+    private static final String APPENDIX = " clicked!";
+    private static final String DATA_OFFSET = "data-offset";
+    private static final String DATA_OFFSET_VALUE = "10,20";
+    private static final String DATA_REFERENCE = "data-reference";
+    private static final String DATA_REFERENCE_VALUE = "parent";
+    private static final String[] OFFSET_DROPDOWN_CONTENTS = new String[]{ITEM_ONE, ITEM_TWO, ITEM_THREE};
+    private static final String[] REFERENCE_DROPDOWN_CONTENTS = new String[]{ITEM_ONE, ITEM_TWO, ITEM_THREE, ITEM_FOUR};
+
+    @DataProvider
+    public Object[][] dropdownData() {
+        return new Object[][]{
+                {offsetDropdown},
+                {referenceDropdown}
+        };
+    }
+
+    @DataProvider
+    public Object[][] dropdownLinkData() {
+        return new Object[][]{
+                {offsetDropdown, OFFSET_DROPDOWN_CONTENTS},
+                {referenceDropdown, REFERENCE_DROPDOWN_CONTENTS}
+        };
+    }
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        bsPage.shouldBeOpened();
+    }
+
+    @Test(dataProvider = "dropdownData")
+    public void expandCollapseTest(Dropdown dropdown) {
+        dropdown.expand();
+        dropdown.is().expanded();
+        dropdown.collapse();
+        dropdown.is().collapsed();
+    }
+
+    @Test(dataProvider = "dropdownLinkData")
+    public void dropdownLinkTest(Dropdown dropdown, String[] linkArray) {
+        for (int i = 1; i <= dropdown.list().size(); i++) {
+            dropdown.expand();
+            dropdown.list().get(i).click();
+            validateAlert(is(linkArray[i - 1] + APPENDIX));
+            dropdown.collapse();
+        }
+    }
+
+    @Test
+    public void dropdownContentTest() {
+        offsetDropdown.expand();
+        offsetDropdown.assertThat()
+                .itemValues(is(Arrays.asList(OFFSET_DROPDOWN_CONTENTS)))
+                .hasItems(ITEM_ONE)
+                .hasItems(ITEM_TWO)
+                .hasItems(ITEM_THREE);
+
+        referenceDropdown.expand();
+        referenceDropdown.assertThat()
+                .itemValues(is(Arrays.asList(REFERENCE_DROPDOWN_CONTENTS)))
+                .hasItems(ITEM_ONE)
+                .hasItems(ITEM_TWO)
+                .hasItems(ITEM_THREE)
+                .hasItems(ITEM_FOUR);
+    }
+
+    @Test
+    public void optionsCssTest() {
+        offsetDropdown.expand();
+        assertThat(offsetDropdown.core().childs().get(1).getAttribute(DATA_OFFSET), is(DATA_OFFSET_VALUE));
+        offsetDropdown.collapse();
+
+        referenceDropdown.expand();
+        assertThat(referenceDropdown.core().childs().get(2).getAttribute(DATA_REFERENCE), is(DATA_REFERENCE_VALUE));
```

Comment:
You should use JDI asserts:
`.assertThat().attr()`

Reply:
Resolved